compiler/src/iree/compiler/DispatchCreation/BubbleUpExpandShapes.cpp

```diff
@@ -38,6 +38,77 @@ struct BubbleUpExpandShapesPass final
   void runOnOperation() override;
 };
 
+// TODO: move this upstream with other tensor bubbling patterns.
+struct BubbleExpandThroughExtract final
+    : public OpRewritePattern<tensor::ExpandShapeOp> {
+
+  using OpRewritePattern<tensor::ExpandShapeOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(tensor::ExpandShapeOp expandOp,
+                                PatternRewriter &rewriter) const override {
+    auto extractOp = expandOp.getSrc().getDefiningOp<tensor::ExtractSliceOp>();
+    if (!extractOp) {
+      return failure();
+    }
+
+    auto origType = extractOp.getSourceType();
```

Comment:
I think using `srcType` (which means that it is the source tensor type) is easier to follow.